### PR TITLE
introduce swar routines for capitalization

### PR DESCRIPTION
**Pull request by anonrig**
Improved the performance of `toUpper` and `toLower` functionality by introducing SWAR routines.

### CODE REVIEW

**Review by jasnell — APPROVED** [review]
Might be good to migrate these into kj

File: src/workerd/api/util.c++ [hunk 1]
```diff
@@ -40,19 +40,61 @@ kj::ArrayPtr<const char> split(kj::ArrayPtr<const char>& text, char c) {
...
+}
+
+// SWAR routine designed to convert ASCII uppercase letters to lowercase.
+// Let's process 8 bytes (64 bits) at a time using a single 64-bit int,
+// treating as 8 parallel 8-bit values.
+// PS: This will enable the use of auto-vectorization.
+constexpr void toLowerAscii(char* input, size_t length) noexcept {
+  uint64_t broadcast_80 = broadcast(0x80);
+  uint64_t broadcast_Ap = broadcast(128 - 'A');
+  uint64_t broadcast_Zp = broadcast(128 - 'Z' - 1);
```
Comment (npaun) [inline src/workerd/api/util.c++:54]: I think the compiler can figure this out, but in principle, shouldn't we mark these as `static constexpr`?

Comment (fhanau) [reply src/workerd/api/util.c++:54]: Note that static or a storage class in general is not needed where there is constexpr; in fact clang-19 warns about `static constexpr` so just constexpr is sufficient.

Comment (npaun) [reply src/workerd/api/util.c++:54]: @fhanau TIL. Thanks!

Comment (anonrig) [reply src/workerd/api/util.c++]: I changed them to `constexpr const`

**Comment by vicb** [comment]
I guess the reason to repeat the same code twice (excluding the 2 boundaries) is perf?

**Comment by vicb** [comment]
Also the one thing I don't get is how overflow is not a problem?

**Pull request merged**